### PR TITLE
SRCH-3351 language support for built-in languages (Hindi and Bengali)

### PR DESCRIPTION
**Pull request by danswick**
## Summary
- Adds language support for content in Hindi and Bengali.
- Adds a new language support example group to the document search spec.
 
### Checklist
Please ensure you have addressed all concerns below before marking a PR "ready for review" or before requesting a re-review. If you cannot complete an item below, replace the checkbox with the ⚠️ `:warning:` emoji and explain why the step was not completed.
 
#### Functionality Checks

- [x] You have run `bundle update` and committed your changes to Gemfile.lock.
 
- [x] You have merged the latest changes from the target branch (usually `main`) into your branch.
  
- [x] Your primary commit message is of the format **SRCH-#### \<description\>** matching the associated Jira ticket. **N/A: "squash and merge" is the default for this repo, so the commit message will match the PR title.**

- [x] PR title is either of the format **SRCH-#### \<description\>** matching the associated Jira ticket (i.e. "SRCH-123 implement feature X"), or **Release - SRCH-####, SRCH-####, SRCH-####** matching the Jira ticket numbers in the release.
 
- [x] Automated checks pass. If Code Climate checks do not pass, explain reason for failures: see [inline comment](https://github.com/GSA/i14y/pull/205/files#r980535559). CC checks are unrelated to this PR's code changes. I am happy to mark as resolved or make a change, but am seeking some feedback. 
 
#### Process Checks

- [x] You have specified at least one "Reviewer".

### CODE REVIEW

File: app/templates/documents.rb [hunk 1]
```diff
@@ -2,8 +2,8 @@
 
 class Documents
   include Templatable
-  LIGHT_STEMMERS = { fr: "french", de: "german", es: "spanish", it: "italian", pt: "portuguese" }
-  STANDARD_STEMMERS = { en: "english", fi: "finnish", hu: "hungarian", ru: "russian", sv: "swedish" }
+  LIGHT_STEMMERS = { fr: 'french', de: 'german', es: 'spanish', it: 'italian', pt: 'portuguese' }
```
Comment (danswick) [inline app/templates/documents.rb:5]: Rubocop's rule, "Prefer single-quoted strings when you don't need string interpolation or special symbols."

File: app/templates/documents.rb [hunk 2]
```diff
@@ -2,8 +2,8 @@
...
+  STANDARD_STEMMERS = { en: 'english', fi: 'finnish', hu: 'hungarian', ru: 'russian', sv: 'swedish', hi: 'hindi', bn: 'bengali' }
```
Comment (danswick) [inline app/templates/documents.rb:6]: Rubocop also flagged these lines as "Freeze mutable objects assigned to constants," but I'm not sure if we want to do that here.

**Comment by danswick** [comment]
@MothOnMars happy to discuss any of the changes here! I made a couple of inline comments about Rubocop issues and am also not totally sure if my approach for testing languages is well-aligned with Ruby conventions.

**Comment by MothOnMars** [comment]
@danswick , I'll start looking at this, but can you update the PR so that all the checklist tasks are completed?

File: spec/classes/document_search_spec.rb [hunk 3]
```diff
@@ -743,4 +743,57 @@ def create_documents(document_hashes)
...
+      lang_code, lang_name, content, query = lang.values_at(:lang_code, :lang_name, :content, :query)
+
+      before do
+        create_documents([
+                           {
+                             language: lang[:lang_code],
+                             created: DateTime.now,
+                             path: "https://vote.gov/#{lang_code}",
+                             title: "title #{lang_name}",
+                             description: "description #{lang_name}",
```
Comment (MothOnMars) [inline spec/classes/document_search_spec.rb:785]: I would remove `created`, `title` and `description`, as these are not being used by the spec.

**Review by MothOnMars — CHANGES_REQUESTED** [review]
@danswick , I have a few requested changes.

For now, I think these specs are sufficient for the purposes of this story. They can be fleshed out and refined as we add more languages, but these test what they need to test, and result in readable spec output, so they are fine for now. 

The structure of the spec (iterating over an assigned array) is a little wonky, but at this point in time, I don't have a better way to do this. We could probably use [shared examples](https://relishapp.com/rspec/rspec-core/docs/example-groups/shared-examples) for this, but I think your version may be more readable. I'll noodle a bit more, but for now let's just stick with what's sufficient and (mostly) done.

File: spec/classes/document_search_spec.rb [hunk 4]
```diff
@@ -743,4 +743,57 @@ def create_documents(document_hashes)
...
+                             content: content
+                           }
+        ])
+      end
+
+      it "gets results for #{lang_name}" do
+        document_search_results = described_class.new(search_options.merge(query: query, language: lang_code)).search
+        expect(document_search_results.total).to eq 1
+        expect(document_search_results.results.first['language']).to eq lang_code
+        expect(document_search_results.results.first['content']).to match(/#{lang[:query]}/)
```
[the remark this answers is not in the document]
Comment (danswick) [reply spec/classes/document_search_spec.rb]: Good point! I think makes sense to reduce this down to just one expectation. The test case is pretty limited and the first two expectations don't really add any useful information as far as I can tell: the value of `content` should break expectations if there are 0 documents and we already have examples for the language parameter/filter.

File: config/initializers/locales.rb [hunk 5]
```diff
@@ -1,5 +1,5 @@
 SUPPORTED_LOCALES = [:ar, :be, :bg, :bn, :ca, :cs, :da, :de, :el, :en, :es, :et, :fa, :fi, :fr, :he, :hi, :hr, :ht,
                      :hu, :hy, :id, :it, :ja, :ka, :km, :ko, :lt, :lv, :mk, :nl, :pl, :ps, :pt, :ro, :ru, :sk, :so,
                      :sq, :sr, :sw, :th, :tr, :uk, :ur, :uz, :vi, :zh]
-LANGUAGE_ANALYZER_LOCALES = [:es, :en, :de, :fi, :fr, :hu, :it, :ja, :ko, :pt, :ru, :sv, :zh]
+LANGUAGE_ANALYZER_LOCALES = [:bn, :de, :en, :es, :fi, :fr, :hi, :hu, :it, :ja, :ko, :pt, :ru, :sv, :zh]
```
Comment (danswick) [inline config/initializers/locales.rb:4]: Only `:hi` and `:bn` were added. The other changes on this line are just from alphabetization.

**Review by MothOnMars — APPROVED** [review]
LGTM

**Pull request merged**